### ISSUE TITLE
How to emit a "warning" of sorts when compiling a document?

### ISSUE THREAD

**Issue by ondohotola** [typst/typst#1322]
### Description

In issue #1276 @Enivex proposed a **very** helpful solution for missing references. 

However, I noticed that that will happily generate **`(?)`** (which I have adapted to **`??`** without reporting this, which means you may end up with orphan references.

I have not been able to find a way to make typst squeal so I can adapt the `#show raw` rule's missing condition.

### Use Case

There should be other conditions where a `warning` type message triggered programatically might be helpful.

**Comment by laurmaedje** [comment]
Related: #600

**Comment by adrfantini** [comment]
Agreed. I programmatically generate Typst manuals and I strive not to make the generation fail, because this will allow me to see what the output looks like - however I **do** want to check the logs and see if I missed some keyword in my index files or if there is a typo, or anything for which I built-in a fallback, but it's not as nice as if it worked properly.

**Comment by Seilerflation** [comment]
Any update on this issue?
